packages/core/src/model.d.ts

```diff
@@ -838,15 +838,16 @@ export type Order = Fn | Col | Literal | OrderItem[];
  */
 export type ProjectionAlias = readonly [string | Literal | Fn | Col | Cast, string];
 
-export type FindAttributeOptions =
+export type FindAttributeOptions<TAttributes = any> =
   | Array<string | ProjectionAlias | Literal>
+  | Array<Extract<keyof TAttributes, string>>
```

Comment:
This is a good change but I wonder if we could go even further by only allowing known attributes when using a string

```suggestion
  | Array<Extract<keyof TAttributes, string> | ProjectionAlias | Literal>
```

If the user needs to reference an unknown attribute, they can still use `col` or [`identifier`](https://github.com/sequelize/sequelize/pull/15598)

Reply:
I was wondering about this and have updated the PR to only known attributes for string values.